exchanges/poloniex/poloniex_futures_types.go

```diff
@@ -0,0 +1,559 @@
+}
+
+// RiskLimit represents a risk limit of futures instrument
+type RiskLimit struct {
+	Symbol                 currency.Pair `json:"symbol"`
+	MarginMode             string        `json:"mgnMode"`
+	Tier                   string        `json:"tier"`
+	MaxLeverage            types.Number  `json:"maxLever"`
+	MaintenanceMarginRatio types.Number  `json:"mMRatio"`
+	MaxSize                types.Number  `json:"maxSize"`
+	MinSize                types.Number  `json:"minSize"`
+}
+
+// WsFuturesCandlesctick represents a kline data for futures instrument
+type WsFuturesCandlesctick struct {
+	Symbol       currency.Pair
+	LowestPrice  types.Number
+	HighestPrice types.Number
+	OpenPrice    types.Number
+	ClosePrice   types.Number
+	Amount       types.Number
+	Quantity     types.Number
+	Trades       types.Number
+	StartTime    types.Time
+	EndTime      types.Time
+	PushTime     types.Time
+}
+
+// UnmarshalJSON deserializes byte data into futures candlesticks into *WsFuturesCandlesctick
+func (o *WsFuturesCandlesctick) UnmarshalJSON(data []byte) error {
+	return json.Unmarshal(data, &[11]any{&o.Symbol, &o.LowestPrice, &o.HighestPrice, &o.OpenPrice, &o.ClosePrice, &o.Amount, &o.Quantity, &o.Trades, &o.StartTime, &o.EndTime, &o.PushTime})
+}
+
+// FuturesTrades represents a futures trades detail
+type FuturesTrades struct {
+	ID           int64         `json:"id"`
+	Timestamp    types.Time    `json:"ts"`
+	Symbol       currency.Pair `json:"s"`
+	Price        types.Number  `json:"px"`
+	Quantity     types.Number  `json:"qty"`
+	Amount       types.Number  `json:"amt"`
+	Side         string        `json:"side"`
+	CreationTime types.Time    `json:"cT"`
+}
+
+// WsFuturesMarkAndIndexPriceCandle represents a websocket k-line data for mark/index candlestick data
+type WsFuturesMarkAndIndexPriceCandle struct {
+	OpeningPrice  types.Number
+	HighestPrice  types.Number
+	LowestPrice   types.Number
+	ClosingPrice  types.Number
+	StartTime     types.Time
+	EndTime       types.Time
+	Symbol        currency.Pair
+	PushTimestamp types.Time
+}
+
+// UnmarshalJSON deserializes byte data into WsFuturesMarkAndIndexPriceCandle instance
+func (v *WsFuturesMarkAndIndexPriceCandle) UnmarshalJSON(data []byte) error {
+	return json.Unmarshal(data, &[8]any{&v.Symbol, &v.LowestPrice, &v.HighestPrice, &v.OpeningPrice, &v.ClosingPrice, &v.StartTime, &v.EndTime, &v.PushTimestamp})
+}
+
+// MarginModeSwitchResponse represents a response detail after switching margin mode for a symbol
+type MarginModeSwitchResponse struct {
+	MarginMode string        `json:"mgnMode"`
+	Symbol     currency.Pair `json:"symbol"`
+}
+
+// ContractLimitPrice holds a contracts highest buy price and lowest sell price limits
+type ContractLimitPrice struct {
+	Symbol    currency.Pair `json:"symbol"`
+	BuyLimit  float64       `json:"buyLmt"`
+	SellLimit float64       `json:"sellLmt"`
+}
+
+// SubscriptionResponse represents a subscription response detail
+type SubscriptionResponse struct {
```

Comment:
🚧 Change request:
You removed the type from `poloniex_types.go` and renamed this one.
Please move this to `poloniex_types.go`